tsdb/store.go

```diff
@@ -135,6 +135,13 @@ type Store struct {
 	baseLogger *zap.Logger
 	Logger     *zap.Logger
 
+	enableStartupProgressMetrics bool
+	startupProgressMetrics       interface {
```

Comment:
I wouldn't have a separate flag to enable startup metrics. If you want them set `startupProgressMetrics`. If not, set it to `nil` or a nop implementation. Having two variables just makes it more cumbersome, and still doesn't address the need for a nil check before `startupProgressMetrics` usages.

Reply:
Sounds good I'll revert this and do nil checking instead 👍